### PR TITLE
Add url encode function

### PR DESCRIPTION
**Pull request by larsalbrecht**
If urls of e.g. episodes contains "ß", the php filter_var function returns false.
So I added a function that encode the url before the check (for further errors) and save this new encodedUrl.

I also cleaned up the code a little bit.

### CODE REVIEW

**Comment by bohwaz** [comment]
Please don't change indentation of changed files, or I won't be able to accept your PR :)

**Comment by larsalbrecht** [comment]
> Please don't change indentation of changed files, or I won't be able to accept your PR :)

You are right, that was not good. I reverted the indentation and added an .editorconfig to define the tab for *.php files.

**Review by bohwaz — COMMENTED** [review]
Except from what I mentioned in the comments, everything is great, thank you :)

File: server/index.php [hunk 1]
```diff
@@ -80,7 +84,7 @@
 	}
 }
 else {
-	echo '<p><a href="login">Login</a></p>';
+	echo '<p><a href="/login">Login</a></p>';
```
[the remark this answers is not in the document]
Comment (larsalbrecht) [reply server/index.php]: Done

File: server/inc/API.php [hunk 2]
```diff
@@ -6,7 +6,7 @@ class API
 	protected ?int $user;
 	protected ?string $section;
 	public ?string $url;
-	public ?string $base_url;
+	private ?string $base_url;
```
[the remark this answers is not in the document]
Comment (larsalbrecht) [reply server/inc/API.php]: Done

File: server/inc/API.php [hunk 3]
```diff
@@ -32,12 +32,19 @@ public function __construct(DB $db)
 		$this->base_url = $url;
 	}
 
-	public function url(string $path = ''): string
+	private function encodeUrl(string $url): string
+	{
+		return preg_replace_callback('#://([^/]+)/([^?]+)#', static function ($match) {
```
[the remark this answers is not in the document]
Comment (larsalbrecht) [reply server/inc/API.php]: done. I add an unparseUrl function to get the correct url back.

**Comment by larsalbrecht** [comment]
You can review again. The conservations are open, so you can add additonal comments if you have anything else :)

**Review by bohwaz — APPROVED** [review]

**Pull request merged**